3D/senior-csharp-test/Assets/Cache/CacheEntry.cs

```diff
@@ -10,6 +10,18 @@ public class CacheEntry<T>
 
 	public CacheEntry( string id, string version, T data, string lastUseDate )
 	{
+		if( string.IsNullOrEmpty( id ) )
+			throw new NullReferenceException( nameof( id ) );
+
+		if( string.IsNullOrEmpty( id ) )
```

Comment:
instead of `id` you should check `version `

Reply:
indeed, fixed